cloud/ec2.go

```diff
@@ -426,6 +427,13 @@ func (m *ec2Manager) spawnOnDemandHost(ctx context.Context, h *host.Host, ec2Set
 			"host_provider": h.Distro.Provider,
 			"distro":        h.Distro.Id,
 		}))
+		detailErr := task.AddHostCreateDetails(h.StartedBy, h.Id, h.SpawnOptions.TaskExecutionNumber, err)
+		grip.ErrorWhen(detailErr != nil, message.WrapError(detailErr, message.Fields{
```

Comment:
ErrorWhen(detailErr != nil) is actually the same as grip.Error(message.WrapError(detailErr, ...) because grip will noop if detailErr is nil.

Reply:
ah thats so smart